util/gen_log_guide.ts

```diff
@@ -0,0 +1,169 @@
+
+type LangObject<T> = { en: T } & { [lang in NonEnLang]?: T };
+
+const curPath = path.resolve();
+
+// Exclude these types since they're not relevant
+type ExcludedLineDocs = 'None';
+type LineDocTypes = Exclude<LogDefinitionTypes, ExcludedLineDocs>;
+
+type LineDocType = {
+  // We can generate `network` type automatically for everything but regex
+  regexes: {
+    network: string;
+    logLine: string;
+  };
+  examples: LangObject<readonly string[]>;
+};
+
+type LineDocs = {
+  [type in LineDocTypes]: LineDocType;
+};
+
+// @TODO: Remove Partial once everything is mapped
+const lineDocs: Partial<LineDocs> = {
+  GameLog: {
+    regexes: {
+      network: NetRegexes.gameLog({ capture: true }).source,
+      logLine: Regexes.gameLog({ capture: true }).source,
+    },
+    examples: {
+      en: [
+        '00|2021-04-26T14:12:30.0000000-04:00|0839||You change to warrior.|d8c450105ea12854e26eb687579564df',
+        '00|2021-04-26T16:57:41.0000000-04:00|0840||You can now summon the antelope stag mount.|caa3526e9f127887766e9211e87e0e8f',
+        '00|2021-04-26T14:17:11.0000000-04:00|0b3a||You defeat the embodiment.|ef3b7b7f1e980f2c08e903edd51c70c7',
+        '00|2021-04-26T14:12:30.0000000-04:00|302b||The gravity node uses Forked Lightning.|45d50c5f5322adf787db2bd00d85493d',
+        '00|2021-04-26T14:12:30.0000000-04:00|322a||The attack misses.|f9f57724eb396a6a94232e9159175e8c',
+      ],
+    },
+  },
+} as const;
+
+type LogGuideOptions = {
+  lang?: string;
+  type?: string;
+};
+
+const isLineType = (type?: string): type is LineDocTypes => {
+  return type !== undefined && type in lineDocs;
+};
+
+const mappedLogLines: LangObject<LineDocTypes[]> = {
+  en: [],
+};
+
+const config: markdownMagic.Configuration = {
+  outputDir: path.posix.relative(curPath, path.posix.join(curPath, 'docs')),
+  transforms: {
+    logLines(_content, options: LogGuideOptions): string {
+      const language = options.lang;
+      const lineType = options.type;
+      if (!isLang(language) || !isLineType(lineType))
+        return '';
+
+      const lineDoc = lineDocs[lineType];
+
+      if (!lineDoc)
+        return '';
+
+      mappedLogLines[language] ??= [];
+      mappedLogLines[language]?.push(lineType);
+
+      const logRepo = new LogRepository();
+      let ret = '';
+      const lineDef = logDefinitions[lineType];
+      const structureNetworkArray = [
+        lineDef.type,
+        '2021-04-26T14:11:35.0000000-04:00',
+      ];
+      let lastIndex = 0;
+
+      for (const [name, index] of Object.entries(lineDef.fields)) {
+        if (['type', 'timestamp'].includes(name))
+          continue;
+        structureNetworkArray[index] = `[${name}]`;
+        lastIndex = Math.max(lastIndex, index);
+      }
+
+      for (let index = 2; index <= lastIndex; ++index)
+        structureNetworkArray[index] ??= '';
+
+      const structureNetwork = structureNetworkArray.join('|');
+      const structureLogLine = ParseLine.parse(logRepo, structureNetwork);
```

Comment:
Oh, neat.  I wonder if this will cause problems with some of the `parseFloat`.  I guess most of the positions aren't included in the act log lines, so it will probably work out.

Reply:
Most if not all of the conversion process should be using the original string and not the `parseXxx` value, so that shouldn't be a problem. If there's anywhere that's not using the original string currently, it should be changed to do so.